rs/rosetta-api/tests/system_tests/test_cases/search_transactions.rs

```diff
@@ -0,0 +1,92 @@
+use crate::common::system_test_environment::RosettaTestingEnvironment;
+use crate::common::utils::test_identity;
+use ic_agent::identity::BasicIdentity;
+use ic_agent::Identity;
+use ic_icrc1_test_utils::{minter_identity, valid_transactions_strategy, DEFAULT_TRANSFER_FEE};
+use icrc_ledger_types::icrc1::account::Account;
+use lazy_static::lazy_static;
+use proptest::strategy::Strategy;
+use proptest::test_runner::Config as TestRunnerConfig;
+use proptest::test_runner::TestRunner;
+use rosetta_core::identifiers::TransactionIdentifier;
+use rosetta_core::request_types::SearchTransactionsRequest;
+use std::sync::Arc;
+use std::time::SystemTime;
+use tokio::runtime::Runtime;
+
+lazy_static! {
+    pub static ref TEST_ACCOUNT: Account = test_identity().sender().unwrap().into();
+    pub static ref MAX_NUM_GENERATED_BLOCKS: usize = 50;
+    pub static ref NUM_TEST_CASES: u32 = 5;
```

Comment:
IIUC, the proptest below runs `NUM_TEST_CASES`, each generating `MAX_NUM_GENERATED_BLOCKS` in the ledger, then syncs rosetta, gets the tip, and searches for the most recent transaction based on the hash of that transaction? Is there a particular reason for generating such a lot of different input, and running the test multiple times? The whole test currently takes over 3 minutes for me to run, which feels like a lot.

In addition to only searching for transactions using the hash of the latest transaction, could you also add test for:

- The hash of the first transaction
- The hash of some other transaction
- A hash that doesn't match any transaction
- No hash at all

Reply:
I can do those tests, note that the test does not yet fully test the entire endpoint. It rather is supposed to test whether the client is working as expected. The full endpoint is tested in various other places throughout the repo. This is why I am trying to consolidate them and move every test one by one into the system test directory. In the process of doing so I would also move all tests that test the search transaction endpoint more thoroughly into this sub directory. 
let me know what you think is the better approach.

Reply:
I"ll reduce the number of test cases.

Reply:
As discussed offline, it doesn't necessarily make sense to test all the functionality at every level - it should be enough to have tests for (some of) the cases I mentioned e.g., at the rosetta level, rather than here at the client level. We can also revisit this once the refactoring and reorganization is done.